### DISCUSSION TITLE
mapping typescript prop   equipment?: Record<string, unknown>;  to MST

### DISCUSSION THREAD

**Discussion by khein-wherewolf** [mobxjs/mobx-state-tree#1750] [Q&A]
I have a typescript interface Guest as so:

interface Guest {
  ...
  equipment?: Record<string, unknown>;
  ...
}

I'm trying to map to MST model:

export const GuestModel = types
  .model("Guest")
  .props({
    ...
    equipment: types.optional(types.map(types.???), {}),
    ...
  })

Looked at types.custom and IAnyType but really haven't found docs or examples that might show how to use?

Any help and/or push in the right direction would be appreciated!

**Comment by YuanRuQian** [comment]
Yeah, I dug through the docs and found nothing... 

It could be really nice to support `Record` out of the box...

Here is my ( very crappy and ugly ) workaround: convert the whole equipment  data to string to store in the props and  then `JSON.parse` it to get the `Record` form from the views

Demo: https://codesandbox.io/s/rxglq?file=/src/index.tsx

<img width="526" alt="Screen Shot 2021-07-31 at 12 17 38" src="https://user-images.githubusercontent.com/32387159/127728364-5dc78dc5-c4bb-4b34-99df-aaeec5191ac4.png">

PS since MST 3+ the `types.map` is wrapped by `types.optional` by default so if you are using a 3+ version you can just write `equipment: types.map(types.???),`

<img width="793" alt="Screen Shot 2021-07-31 at 12 19 49" src="https://user-images.githubusercontent.com/32387159/127728437-64af9d29-6bf0-4b38-b24a-b1012f85748a.png">

**Reply by khein-wherewolf** [reply]
Thanks, and also thanks for pointing out the array/map defaults!
